packages/aws-rfdk/lib/lambdas/nodejs/asg-attach-eni/test/test.ts

```diff
@@ -220,3 +220,49 @@ test('continues when complete lifecycle errors', async () => {
   await expect(handler(event)).resolves.not.toThrow();
   expect(console.error).toHaveBeenCalledTimes(4); // 4 = each of the two records printing two error messages
 });
+
+test('continues when complete lifecycle errors non-error thrown', async () => {
+  // GIVEN
+  const event = {
+    Records: [
+      {
+        Sns: {
+          Message: JSON.stringify({
+            LifecycleTransition: 'autoscaling:EC2_INSTANCE_LAUNCHING',
+            AutoScalingGroupName: 'ASG-Name-1',
+            LifecycleHookName: 'Hook-Name-1',
+            EC2InstanceId: 'i-0000000000',
+            LifecycleActionToken: 'Action-Token-1',
+            NotificationMetadata: JSON.stringify({
+              eniId: 'eni-000000000',
+            }),
+          }),
+        },
+      },
+      {
+        Sns: {
+          Message: JSON.stringify({
+            LifecycleTransition: 'autoscaling:EC2_INSTANCE_LAUNCHING',
+            AutoScalingGroupName: 'ASG-Name-1',
+            LifecycleHookName: 'Hook-Name-1',
+            EC2InstanceId: 'i-0000000000',
+            LifecycleActionToken: 'Action-Token-1',
+            NotificationMetadata: JSON.stringify({
+              eniId: 'eni-000000000',
+            }),
+          }),
+        },
+      },
+    ],
+  };
+
+  attachSpy = jest.fn( (request) => successRequestMock(request) );
+  mock('EC2', 'attachNetworkInterface', attachSpy);
+
+  JSON.parse = jest.fn( () => {throw 47;} );
```

Comment:
Unless I'm missing something, I think we may not be restoring `JSON.parse` after the test. According to [this SO answer](https://stackoverflow.com/a/76292208), we should be able to do:


```suggestion
  jest.spyOn(JSON, 'parse').mockImplementationOnce(jest.fn( () => {throw 47;} ));
```

Reply:
Gotcha, I'm not super familiar with jest. But this makes sense to me.